tf_quant_finance/volatility/bachelier_tf.py

```diff
@@ -0,0 +1,248 @@
+'ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+'OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+'THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+@author: Joerg Kienitz
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+import tensorflow_probability as tfp
+
+# straight fwd implementation of the Bachelier pricing
+# there is a version with just one call to exp !!
+
+    
+def option_price(spots, 
+                 strikes, 
+                 volatilities, 
+                 expiries,
+                 rates,
+                 is_call_options=None,
+                 dtype = None,
+                 name = None):
+    """ Compute the Bachelier price for a batch of European options.
+    
+  ## References:
+  [1] Kienitz, J. "Interest Rate Derivatives Explained I", Plagrave McMillan (2014) p.119
+  [2] https://papers.ssrn.com/sol3/papers.cfm?abstract_id=3428994
+    
+    Parameters
+    ----------
+    spots : A real `Tensor` of any shape. The current spot prices to
+      expiry.
+    strikes : A real `Tensor` of the same shape and dtype as `spots`. The
+      strikes of the options to be priced.
+    volatilities : A real `Tensor` of same shape and dtype as `spots`. The
+      volatility to expiry.
+    expiries : A real `Tensor` of same shape and dtype as `spots`. The expiry
+      for each option. The units should be such that `expiry * volatility**2` is
+      dimensionless.
+    rates : A real `Tensor` of the same shape and dtype as `spots`. The
+      rates of the options to be priced.
+    is_call_options: A boolean `Tensor` of a shape compatible with `forwards`.
+      Indicates whether to compute the price of a call (if True) or a put (if
+      False). If not supplied, it is assumed that every element is a call.
+    dtype: Optional `tf.DType`. If supplied, the dtype to be used for conversion
+      of any supplied non-`Tensor` arguments to `Tensor`.
+      Default value: None which maps to the default dtype inferred by TensorFlow
+      (float32).
+    name: str. The name for the ops created by this function.
+      Default value: None which is mapped to the default name `option_price`.
+
+    Returns
+    -------
+    option_prices: A `Tensor` of the same shape as `spots`. The Bachelier
+    price of the options.
+    
+    #### Examples
+    ```python
+    spots = np.array([0.03, 0.02])
+    strikes = np.array([.02, .02])
+    volatilities = np.array([.004, .005])
+    expiries = 2.0
+    rates = [0.02, 0.01]
+    computed_prices = option_price(
+      spots,
+      strikes,
+      volatilities,
+      expiries,
+      rates,
+      dtype=tf.float64)
+    # Expected print output of computed prices:
+    # <tf.Tensor: id=2482, shape=(2,), dtype=float32, numpy=array([0.01605039, 0.00720789], dtype=float32)>
```

Comment:
This output is suspicious as we would expect `float64`.
Also, let's leave only `[0.01605039, 0.00720789]` since some users might be using TensorFlow 1 and which means they need to follow extra steps before getting the result